onnxruntime/core/providers/shared/utils/utils.cc

```diff
@@ -53,7 +55,11 @@ bool GetClipMinMax(const InitializedTensorSet& initializers, const Node& node,
         return false;
       }
       Initializer unpacked_tensor(*initializers.at(min_name));
-      min = unpacked_tensor.DataAsSpan<float>()[0];
+      if (input_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT) {
+        min = (unpacked_tensor.DataAsSpan<MLFloat16>()[0]).ToFloat();
```

Comment:
>(unpacked_tensor.DataAsSpan<MLFloat16>()[0]).ToFloat() [](http://example.com/codeflow?start=14&length=54)

[minor] Are the extra parens needed?

`(unpacked_tensor.DataAsSpan<MLFloat16>()[0]).ToFloat()`
vs
`unpacked_tensor.DataAsSpan<MLFloat16>()[0].ToFloat()`

Reply:
that was a typo - fixed now.

Reply:
I noticed we don't check that min/max are actually a scalar (and nonzero size), but then I saw the dereference of `unpacked_tensor.DataAsSpan<float>()[0]` should be fine because `gsl::span` would throw if the span was empty.

Reply:
I also noticed we don't check that min/max are actually a scalar (and nonzero size), but then I saw the dereference of `unpacked_tensor.DataAsSpan<float>()[0]` should be fine because `gsl::span` would throw if the span was empty.